src/daylight-savings/range-helpers.ts

```diff
@@ -13,31 +13,36 @@ function parseDateInZone(date: string, timezone: string | null): DateTime {
 
 function getDSTRanges(dstStart: DateTime, dstEnd: DateTime, country: string) {
   const ranges = [];
+
   if (dstStart.year === dstEnd.year) {
     ranges.push([
       `DST Period ${dstStart.year}`,
-      `${dstStart.toFormat(formatString)}`,
-      `${dstEnd.toFormat(formatString)}`,
+      dstStart.toFormat(formatString),
+      dstEnd.toFormat(formatString),
     ]);
   } else {
-    const prevYearDstEnd = parseDateInZone(
-      getDSTEnd(country, dstStart.year - 1) as string,
-      dstEnd.zoneName
-    ); //@todo fix when getDSTEnd returns null
+    const prevYearDstEndString = getDSTEnd(country, dstStart.year - 1);
+    const prevYearDstStartString = getDSTStart(country, dstStart.year - 1);
 
-    const prevYearDstStart = parseDateInZone(
-      getDSTStart(country, dstStart.year - 1) as string,
-      dstStart.zoneName
-    ); //@todo fix when getDSTEnd returns null
+    if (prevYearDstEndString && prevYearDstStartString) {
+      const prevYearDstEnd = parseDateInZone(
+        prevYearDstEndString,
+        dstEnd.zoneName
+      );
+      const prevYearDstStart = parseDateInZone(
+        prevYearDstStartString,
+        dstStart.zoneName
+      );
 
-    ranges.push([
-      `DST Period ${prevYearDstStart.monthShort}, ${prevYearDstStart.year} - ${prevYearDstEnd.monthShort}, ${prevYearDstEnd.year}`,
-      prevYearDstStart.toFormat(formatString),
-      prevYearDstEnd.toFormat(formatString),
-    ]);
+      ranges.push([
+        `DST Period ${prevYearDstStart.monthLong}, ${prevYearDstStart.year} - ${prevYearDstEnd.monthLong}, ${prevYearDstEnd.year}`,
+        prevYearDstStart.toFormat(formatString),
+        prevYearDstEnd.toFormat(formatString),
+      ]);
+    }
 
     ranges.push([
-      `DST Period ${dstStart.monthShort}, ${dstStart.year} - ${dstEnd.monthShort}, ${dstEnd.year}`,
+      `DST Period ${dstStart.monthLong}, ${dstStart.year} - ${dstEnd.monthLong}, ${dstEnd.year}`,
```

Comment:
Technically this is a breaking change. Consider a major version update. Why do we need to make it long form?

Reply:
I thought that it was the most elegant way rather than having a custom method but I could create a method just to do search and replace

Reply:
IIRC, it was short on purpose
1) it's used by editors, mosque admins
2) the modal is already congested and have short spaces. 

i think it's irrelevant to your fix. so may be create a separate PR to test it well. Fow now just adding the dates are good enough. Please also double check those dates!